cupy/random/_generator_api.pyx

```diff
@@ -195,6 +222,40 @@ class Generator:
         # omitted args.
         return (<object>y).astype(dtype, copy=False)
 
+    def gamma(self, shape, scale=1.0, size=None):
+        """Returns an array of samples drawn from a gamma distribution.
+
+        .. seealso::
+            - :func:`cupy.random.gamma` for full documentation
+            - :meth:`numpy.random.RandomState.gamma`
+        """
+        return self.standard_gamma(shape, size) * scale
+
+    def standard_gamma(self, shape, size=None, dtype=numpy.float64, out=None):
+        """Returns an array of samples drawn from a standard gamma distribution.
+
+        .. seealso::
+            - :func:`cupy.random.standard_gamma` for full documentation
+            - :meth:`numpy.random.RandomState.standard_gamma`
+        """
+        cdef ndarray y
+
+        if shape < 0.0:
+            raise ValueError('shape < 0')
+
+        if out is not None:
+            self._check_output_array(dtype, size, out)
+
+        y = ndarray(size if size is not None else (), numpy.float64)
+        _launch_dist(self.bit_generator, standard_gamma, y, (shape,))
+        if out is not None:
+            out[...] = y
+            y = out
```

Comment:
Silly question: Could you remind me why here and elsewhere in this file we had to create a new array `y` first, and then copy the outcome to `out` if provided? Can't we pass `out` to `_launch_dist` directly if it's not `None`, saving one copy?

Reply:
this is a very good question.
Arrays are generated using fp64 due to curand limitations, so we cant write them directly to other types.
But I agree that if the type of out is the same type than the generated array we should pass out directly.
I will fix this once all the random outgoing prs are merged.

Reply:
Fixed :P